lib/hobbes/boot/amapping.hob

```diff
@@ -45,33 +45,57 @@ instance (p++s=a, Function f p rf, Function rf s r) => Function f a r where
 /*
   abstract over "array-ness" (abstracting pattern-matches over 'array-like' things)
 */
-class Array a i | a -> i where
-  size    :: a -> long
-  element :: (a, long) -> i
+class Array xs x | xs -> x where
+  size     :: xs -> long
+  element  :: (xs, long) -> x
+  elements :: (xs, long, long) -> [x]
+
+// a default implementation of 'elements' in terms of 'element' (good enough if indexing is fast)
+instance Convert long long where
```

Comment:
I noticed there exists an [instance in convert.hob](https://github.com/Morgan-Stanley/hobbes/blob/2655bca00e7cd95a9c5561d8f568e71ad09df579/lib/hobbes/boot/convert.hob#L37) dealing with the same type already.  Instead of explicitly defined here, would that help?

Reply:
It's just here because it's implicitly used by array indexing, but possibly all of the 'Convert' machinery should come first.  In any case, this instance has to come before it's used in the following 'elements' implementations.